src/core/dev_api/openvino/runtime/shared_buffer.hpp

```diff
@@ -28,4 +28,56 @@ class SharedBuffer : public ov::AlignedBuffer {
     T _shared_object;
 };
 
+/// \brief SharedStreamBuffer class to store pointer to pre-acclocated buffer and provide streambuf interface.
+///  Can return ptr to shared memory and its size
+class SharedStreamBuffer : public std::streambuf {
+public:
+    SharedStreamBuffer(char* data, size_t size) : m_data(data), m_size(size), m_offset(0) {}
+
+    // get data ptr and its size
+    char* data() {
+        return m_data;
+    }
+
+    size_t size() {
+        return m_size;
+    }
+
+protected:
+    // override std::streambuf methods
+    std::streamsize xsgetn(char* s, std::streamsize count) override {
+        auto real_count = std::min<std::streamsize>(m_size - m_offset, count);
+        std::memcpy(s, m_data + m_offset, real_count);
```

Comment:
Why not use a raw pointer without copy? We need to reed only from buffer during model compilation.
Anyway, please use cpu_parallel_memcpy.

Reply:
std::streambuf::xsgetn use already allocated memory (char* s) to copy data to it. So we have to copy here